### PR TITLE
Use dedicated UnixUserID and UnixGroupID types

### PR DESCRIPTION
**Pull request by jamiehannaford**
**What this PR does / why we need it**:

DRYs up type definitions by using the dedicated types in apimachinery 

**Which issue this PR fixes**

#38120

**Release note**:

UIDs and GIDs now use apimachinery types

### CODE REVIEW

**Comment by smarterclayton** [comment]
@k8s-bot ok to test

**Comment by smarterclayton** [comment]
API change in general approved as per linked thread.

**Comment by jamiehannaford** [comment]
@smarterclayton @yifan-gu @pmorie This is RFR. I know it's a ginormous changeset, but these API types touched lots of the codebase. If there's anything I can do to make reviewing easier (like breaking stuff up) please let me now!

**Comment by jamiehannaford** [comment]
@smarterclayton @yifan-gu Is there any way to make this easier to review? Otherwise I might just close this PR, since it's unlikely folks with have time to review +100 files.

File: pkg/kubelet/volumemanager/volume_manager_test.go [hunk 1]
```diff
@@ -240,7 +241,7 @@ func createObjects() (*v1.Node, *v1.Pod, *v1.PersistentVolume, *v1.PersistentVol
 				},
 			},
 			SecurityContext: &v1.PodSecurityContext{
-				SupplementalGroups: []int64{555},
+				SupplementalGroups: []kubetypes.UnixGroupID{kubetypes.UnixGroupID(555)},
```
Comment (thockin) [inline pkg/kubelet/volumemanager/volume_manager_test.go]: do you need the inner cast?  555 is an untyped constantm shouldn't it convert?

File: pkg/securitycontext/util_test.go [hunk 2]
```diff
@@ -120,7 +121,7 @@ func TestHaRootUID(t *testing.T) {
 }
 
 func TestHasRunAsUser(t *testing.T) {
-	var runAsUser int64 = 0
+	runAsUser := types.UnixUserID(0)
```
Comment (thockin) [inline pkg/securitycontext/util_test.go]: In the topic of making it easier to review, this makes it harder.  Trivial changes are easy to scan.  The requires reading.

**Review by thockin — COMMENTED** [review]
This was not unreviewable.  You might have been able to do it one api-group at a time.  Or to make all your changes in one commit and put all the generated junk in a 2nd commit.

File: pkg/kubelet/volumemanager/volume_manager_test.go (hunk 1 above)
Comment (jamiehannaford) [inline pkg/kubelet/volumemanager/volume_manager_test.go]: Yep, you're right

**Comment by jamiehannaford** [comment]
@thockin Thanks for the review. All tests are now passing :)

**Comment by thockin** [comment]
I'm going to flip this one to reviewable to finish

**Comment by thockin** [comment]
Reviewed 103 of 121 files at r1, 18 of 20 files at r2.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[pkg/api/validation/validation_test.go, line 3303 at r2](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/44714#-KjJKsBUQ16q3osMlCrc:-KjJKsBUQ16q3osMlCrd:b-6sq3hb) ([raw file](https://github.com/kubernetes/kubernetes/blob/22110fd294e4957d2ff1fa071bebf15d89981cbe/pkg/api/validation/validation_test.go#L3303)):*
> ```Go
> 			SecurityContext: &api.PodSecurityContext{
> 				HostNetwork:        false,
> 				SupplementalGroups: []types.UnixGroupID{maxGroupID},
> ```

Why does this series drop the "1234" value?

---

*[pkg/api/validation/validation_test.go, line 8785 at r2](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/44714#-KjJL35i6E5VqhxH3Dny:-KjJL35i6E5VqhxH3Dnz:b-nmv99z) ([raw file](https://github.com/kubernetes/kubernetes/blob/22110fd294e4957d2ff1fa071bebf15d89981cbe/pkg/api/validation/validation_test.go#L8785)):*
> ```Go
> 
> 	negativeRunAsUser := fullValidSC()
> 	negativeUser := types.UnixUserID(-1)
> ```

again in the topic of simpler reviews - changing the style of these made it net harder.

---

*[pkg/apis/extensions/validation/validation_test.go, line 2351 at r2](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/44714#-KjJLLRqr_b-Qoo8W81r:-KjJLLRrMrVD7_7x8ayO:byzm50q) ([raw file](https://github.com/kubernetes/kubernetes/blob/22110fd294e4957d2ff1fa071bebf15d89981cbe/pkg/apis/extensions/validation/validation_test.go#L2351)):*
> ```Go
> 	invalidUIDPSP.Spec.RunAsUser.Rule = extensions.RunAsUserStrategyMustRunAs
> 	invalidUIDPSP.Spec.RunAsUser.Ranges = []extensions.UserIDRange{
> 		{Min: types.UnixUserID(-1), Max: types.UnixUserID(1)},
> ```

shouldn't need the explicit casts here and the rest of this file?

---

*[test/e2e/security_context.go, line 69 at r2](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/44714#-KjJNrSqcuNY0LD9IkbZ:-KjJNrSqcuNY0LD9Ikb_:b-1hovln) ([raw file](https://github.com/kubernetes/kubernetes/blob/22110fd294e4957d2ff1fa071bebf15d89981cbe/test/e2e/security_context.go#L69)):*
> ```Go
> 		pod := scTestPod(false, false)
> 		pod.Spec.Containers[0].Command = []string{"id", "-G"}
> 		pod.Spec.SecurityContext.SupplementalGroups = []types.UnixGroupID{types.UnixGroupID(1234), types.UnixGroupID(5678)}
> ```

shouldn't need a cast here

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/44714)*

**Comment by yifan-gu** [comment]
I am probably not the best person to reviewing changes here, reassigning to Tim.

**Comment by jamiehannaford** [comment]
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[pkg/api/validation/validation_test.go, line 3303 at r2](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/44714#-KjJKsBUQ16q3osMlCrc:-KjN6-X7LYarZAZS8bwe:b-piqxju) ([raw file](https://github.com/kubernetes/kubernetes/blob/22110fd294e4957d2ff1fa071bebf15d89981cbe/pkg/api/validation/validation_test.go#L3303)):*
<details><summary><i>Previously, thockin (Tim Hockin) wrote…</i></summary><blockquote>

Why does this series drop the "1234" value?
</blockquote></details>

Good catch - added back in.

---

*[pkg/apis/extensions/validation/validation_test.go, line 2351 at r2](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/44714#-KjJLLRqr_b-Qoo8W81r:-KjN64Y59cOIrTnR5t3i:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/22110fd294e4957d2ff1fa071bebf15d89981cbe/pkg/apis/extensions/validation/validation_test.go#L2351)):*
<details><summary><i>Previously, thockin (Tim Hockin) wrote…</i></summary><blockquote>

shouldn't need the explicit casts here and the rest of this file?
</blockquote></details>

Done.

---

*[pkg/securitycontext/util_test.go, line 124 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/44714#-Kiej0ELCT9mUIXbFpeP-r1-124:-KjN65yqvqwecJUFQ2Ae:b-lvv8qa) ([raw file](https://github.com/kubernetes/kubernetes/blob/710d141c68706e25bf1eea848591cc6362ee995b/pkg/securitycontext/util_test.go#L124)):*
<details><summary><i>Previously, thockin (Tim Hockin) wrote…</i></summary><blockquote>

In the topic of making it easier to review, this makes it harder.  Trivial changes are easy to scan.  The requires reading.
</blockquote></details>

Yeah, for future reference I'll try to reduce the delta as much as possible. Thanks for the tip :smile: 

---

*[test/e2e/security_context.go, line 69 at r2](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/44714#-KjJNrSqcuNY0LD9IkbZ:-KjN5uZT_Ld5Pg_hrQLp:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/22110fd294e4957d2ff1fa071bebf15d89981cbe/test/e2e/security_context.go#L69)):*
<details><summary><i>Previously, thockin (Tim Hockin) wrote…</i></summary><blockquote>

shouldn't need a cast here
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/44714)*

**Comment by jamiehannaford** [comment]
Hm, strange. The bazel build is failing, but when you click on the Details link it says `Unable to load build details`. When I run `hack/verify-bazel.sh` locally, no errors are reported, so everything seems up to date.

@thockin Any ideas what may be causing this? I've tried re-pushing and the same thing occurs.

**Comment by thockin** [comment]
@k8s-bot bazel test this

**Comment by thockin** [comment]
/lgtm

---







*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/44714#-:-KjO9g4XfV9FKAZGR5JV:b-8ks5qa)*

**Comment by thockin** [comment]
@ixdy - bazel build fails immediately, no data.

**Comment by ixdy** [comment]
issue with prow.
@k8s-bot bazel test this

**Comment by lavalamp** [comment]
@smarterclayton, @thockin: This PR included non-api types in the api type tree--that would have been good to call out in the review.

**Comment by smarterclayton** [comment]
Hrm, we already have types.UID

**Comment by lavalamp** [comment]
I'm adding openapi definitions because that seems easier than unbreaking this.

**Comment by lavalamp** [comment]
Workaround: #47753

**Comment by mbohlool** [comment]
Look like the OpenAPI changes is not the source of compatibility issue. It should be an old swagger 1.2 change. I will look into a solution.

On a separate note, on things to call out in the review, can we make it a requirement to have separate commits for generated code. It is hard to find manual code inside this PR (I assume all types.go files, but I cannot be sure).

**Comment by caesarxuchao** [comment]
I removed the release note because the PR is reverted

**Pull request merged**